src/core/openthread-thread_v1_2-core-default-config.h

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes default compile-time configuration constants
+ *   for Thread 1.2 in OpenThread's implementation.
+ */
+
+#ifndef OPENTHREAD_THREAD_V1_2_CORE_DEFAULT_CONFIG_H_
+#define OPENTHREAD_THREAD_V1_2_CORE_DEFAULT_CONFIG_H_
+
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+
+/**
+ * @def OPENTHREAD_CONFIG_DEFAULT_CSL_SYNCHRONIZED_TIMEOUT
+ *
+ * The default CSL sample duration in seconds.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_DEFAULT_CSL_SYNCHRONIZED_TIMEOUT
+#define OPENTHREAD_CONFIG_DEFAULT_CSL_SYNCHRONIZED_TIMEOUT 100
```

Comment:
The common naming convension is that we try to have all the config names in a `src/core/config/{module}.h` include the `{module}` name immediately after the `OPENTHREAD_CONFIG_` (e.g. you check `config/mac.h`)
So suggest naming this `OPENTHREAD_CONFIG_CSL_DEFAULT_SYNCHRONIZED_TIMEOUT`